### PR TITLE
Add new restrictions to CancelJob

### PR DESCRIPTION
**Pull request by pecanoro**
@mea36 

Like we discussed in the original issue, I added two new restrictions:

- CancelJob can only be called on a job with a parent
- CancelJob can only be called on child job that has at least one sibling in the RUNNING state

### Issues

$ https://github.com/Expensify/Expensify/issues/60822

### Tests

Automated

### CODE REVIEW

File: plugins/Jobs.cpp [hunk 1]
```diff
@@ -319,6 +324,21 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
             return true; // Done
         }
 
+        // We can't control where CancelJob is called from, but we can verify that there's at least one RUNNING sibling around
+        if (!db.read("SELECT j.jobID "
```
Comment (mea36) [inline plugins/Jobs.cpp]: We know the parentJobID from the query above.
We can simplify this to 
```sql
SELECT jobID
FROM jobs
WHERE parentID = result[0][2] and state = 'RUNNING'
LIMIT 1;
```

File: test/tests/jobs/CancelJobTest.cpp [hunk 2]
```diff
@@ -84,18 +85,38 @@ struct CancelJobTest : tpunit::TestFixture {
         tester->executeWaitVerifyContent(command, "404 Invalid jobID - Cannot cancel a job with children");
     }
 
-    // Ignore canceljob for RUNNING jobs
+    // Ignore canceljob of RUNNING jobs
```
Comment (mea36) [inline test/tests/jobs/CancelJobTest.cpp]: Can you change this back to `for`?

File: test/tests/jobs/CancelJobTest.cpp [hunk 3]
```diff
@@ -206,28 +227,86 @@ struct CancelJobTest : tpunit::TestFixture {
...
-        // Cancel it
+        // Create a sibling
+        command.clear();
+        command.methodLine = "CreateJob";
+        command["name"] = "sibling";
+        command["parentJobID"] = parentID;
+        response = tester->executeWaitVerifyContentTable(command);
+        string siblingID = response["jobID"];
+
+        // Finish the parent to put the child in the QUEUED state
```
Comment (mea36) [inline test/tests/jobs/CancelJobTest.cpp]: `to put the children in `

File: test/tests/jobs/CancelJobTest.cpp [hunk 4]
```diff
@@ -206,28 +227,86 @@ struct CancelJobTest : tpunit::TestFixture {
...
+        command.methodLine = "FinishJob";
+        command["jobID"] = parentID;
+        tester->executeWaitVerifyContent(command);
+
+        // Try cancelling one of the children, but none of them are running so it should fail.
         command.clear();
         command.methodLine = "CancelJob";
-        command["jobID"] = jobID;
+        command["jobID"] = childID;
+        tester->executeWaitVerifyContent(command, "404 Invalid jobID - Cannot cancel a job that has no RUNNING siblings");
```
Comment (mea36) [inline test/tests/jobs/CancelJobTest.cpp]: You don't need to test this here, you test this in `cancelJobWithoutSiblings`

File: test/tests/jobs/CancelJobTest.cpp (hunk 2 above)
Comment (pecanoro) [reply test/tests/jobs/CancelJobTest.cpp]: No idea, probably I wanted to write something else and wrote that when I undid it

File: test/tests/jobs/CancelJobTest.cpp (hunk 4 above)
Comment (pecanoro) [reply test/tests/jobs/CancelJobTest.cpp]: Copy/paste lol

**Comment by pecanoro** [comment]
Updated!

**Comment by pecanoro** [comment]
Updated!

**Pull request merged**